lib/mrb/scripts/command_line/grndb.rb

```diff
@@ -402,6 +405,37 @@ def failed_to_open(name)
           failed(message)
         end
       end
+
+      class Recoverer
+        attr_writer :database
+
+        def initialize
+          @context = Context.instance
+        end
+
+        def recover(force_truncate)
+          @database.recover
+          if force_truncate
+            @database.each do |object|
+              if object.corrupt?
+                logger = @context.logger
+                Dir.foreach(File.dirname(object.path)) do |path|
+                  if path =~ /^#{File.basename(object.path)}.+/
+                    begin
+                      File.unlink("#{File.dirname(object.path)}/#{path}")
+                      message = "Removed file which is related to corrupted #{object.path}: #{path}"
+                      logger.log(Logger::Level::INFO.to_i, __FILE__, __LINE__, "recover", message)
```

Comment:
Users will not notice the information with this code. Because the default `grndb` log path is invalid for normal users and the `info` level log isn't logged by default.

So I don't agree with

> standard output may be lost by user without redirecting

The code does nothing on many cases. I think that it's danger than standard output without redirect.

If you want to log, you should do both printing to standard output and logging.